src/raftstore/store/peer.rs

```diff
@@ -1177,20 +1166,26 @@ impl Peer {
         let last_pending_read_count = self.raft_group.raft.pending_read_count();
         let last_ready_read_count = self.raft_group.raft.ready_read_count();
 
-        self.raft_group.read_index(uuid.as_bytes().to_vec());
+        let id = self.pending_reads.id_allocator;
+        // It's ok to overflow. If there are two entries share same id, then
+        // there should be u64::MAX entries kept in memory in which case TiKV
+        // should oom already. At lease true for amd64 architect.
+        self.pending_reads.id_allocator = id.overflowing_add(1).0;
```

Comment:
Please abstract alloc method for id allocator, and use it instead of overflowing_add.